File: solutions/typescript/124-valid-anagram.ts
/**
 * Problem 124: Valid Anagram (LeetCode 242)
 * Difficulty: Easy
 * Language: TypeScript
 */
function isAnagram(s: string, t: string): boolean {
  const freq = (str: string) =>
    [...str].reduce<Record<string, number>>(
      (acc, c) => ({ ...acc, [c]: (acc[c] || 0) + 1 }), {});
  const fs = freq(s);
  const ft = freq(t);
  return JSON.stringify(Object.entries(fs).sort()) ===
         JSON.stringify(Object.entries(ft).sort());
}

File: solutions/ocaml/124-valid-anagram.ml
(*
  Problem 124: Valid Anagram (LeetCode 242)
  Difficulty: Easy
  Language: OCaml
*)
let is_anagram s t =
  let sorted str =
    String.to_seq str |> List.of_seq |> List.sort Char.compare in
  sorted s = sorted t
